pkg/operator/controller/dns/controller_test.go

```diff
@@ -840,3 +847,26 @@ func Test_customCABundle(t *testing.T) {
 		})
 	}
 }
+
+func buildFakeCache(t *testing.T) *testutil.FakeCache {
+	t.Helper()
+	scheme := runtime.NewScheme()
+	iov1.AddToScheme(scheme)
+	fakeClient := fake.NewClientBuilder().
+		WithScheme(scheme).
+		WithIndex(&iov1.DNSRecord{}, dnsRecordIndexFieldName, func(o client.Object) []string {
```

Comment:
nit (no need to fix now!): do we care about making this indexer function some sort of utils/specific function that can be used both on the `operatorCache.IndexField` and on fakeCache to keep consistency?

Reply:
Do you mean a util function that the controller logic and test logic would share?  That does make sense, though I do caution against re-using controller logic in tests if doing so could mask a defect in the controller logic.